agent/node.go

```diff
@@ -74,6 +74,11 @@ type NodeConfig struct {
 	// HeartbeatTick defines the amount of ticks between each
 	// heartbeat sent to other members for health-check purposes
 	HeartbeatTick uint32
+
+	// Spec is the spec of the node being requested, not all fields in
```

Comment:
Isn't this a bit confusing since a `Spec` should always be honored by definition?